### PR TITLE
Introduce yapf as project formatting standard

### PR DESCRIPTION
**Pull request by Achimh3011**
In order to make the formatting rules for the project a little more succint, I propose to employ yapf.

Not complying to yapf would be a reason to reject a PR (until fixed), however there is no automated test for now.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[yapf]
+based_on_style = pep8
+spaces_before_comment = 4
+split_before_logical_operator = true
+column_limit = 120
```
Comment (nwilbert) [inline setup.cfg]: I would prefer to stick to the suggested PEP8 standard of 79/80 characters. Note that the source files already conformed to this line length.

Personally I quite like the readability benefits of a short line length (as it encourages semantic indentation). If it is good enough for the standard library it should be good for everyone ;-).

I know this topic is super controversial...

File: src/jiradeps/graphvizdeps.py [hunk 2]
```diff
@@ -10,13 +9,16 @@
...
-def visualize_with_graphviz(nx_graph: nx.DiGraph, epics,
-                            file_name=None, file_type='svg', group_epics=False,
-                            align_sprints=False, show_status=False):
+def visualize_with_graphviz(nx_graph: nx.DiGraph,
+                            epics,
+                            file_name=None,
+                            file_type='svg',
+                            group_epics=False,
+                            align_sprints=False,
+                            show_status=False):
```
Comment (nwilbert) [inline src/jiradeps/graphvizdeps.py]: Strange, I think the original indentation was already PEP8 compliant. The new one doesn't look like an improvement.

File: src/jiradeps/graphvizdeps.py [hunk 3]
```diff
@@ -48,8 +49,8 @@ def visualize_with_graphviz(nx_graph: nx.DiGraph, epics,
         else:
             sprint_display_number = '?'
             color = 'black'
-        label = _get_node_label(issue, issue_sprints, sprint_display_number,
-                                data['max_pred_dist'], show_status=show_status)
+        label = _get_node_label(
+            issue, issue_sprints, sprint_display_number, data['max_pred_dist'], show_status=show_status)
```
Comment (nwilbert) [inline src/jiradeps/graphvizdeps.py]: Again, I don't think that this is an improvement.

**Comment by nwilbert** [comment]
I'm not entirely happy with the results, but this is probably mostly an effect of the increased line length. I'll take a deeper look at it on Friday.

File: src/jiradeps/graphvizdeps.py (hunk 3 above)
Comment (nwilbert) [inline src/jiradeps/graphvizdeps.py]: This is another change that I don't really like.

File: src/jiradeps/graphvizdeps.py [hunk 4]
```diff
@@ -65,35 +66,28 @@ def visualize_with_graphviz(nx_graph: nx.DiGraph, epics,
             if align_sprints:
                 sub_graph.body += [
                     '\t{{rank=same;{}}}'.format(' '.join(sprint_nodes))
-                    for sprint_nodes in epic_sprint_nodes[epic_key].values()]
+                    for sprint_nodes in epic_sprint_nodes[epic_key].values()
+                ]
```
Comment (nwilbert) [inline src/jiradeps/graphvizdeps.py]: dislike

File: src/jiradeps/main.py [hunk 5]
```diff
@@ -15,46 +14,39 @@
...
-@click.option('--configfile', '-c',
-              default=os.path.join(os.getcwd(), CONFIG_FILENAME),
-              help='Config file to be used '
-                   f'(defaults to CWD/{CONFIG_FILENAME})')
+@click.option(
+    '--configfile',
+    '-c',
+    default=os.path.join(os.getcwd(), CONFIG_FILENAME),
+    help='Config file to be used '
+    f'(defaults to CWD/{CONFIG_FILENAME})')
```
Comment (nwilbert) [inline src/jiradeps/main.py]: dislike

**Comment by Achimh3011** [comment]
OK, I see. I'll try to tweak settings to get the formatting better some time. Until then I close the PR.

**Pull request closed without merging**